### DISCUSSION TITLE
Using secret in KafkaConnector config

### DISCUSSION THREAD

**Discussion by m-kay** [strimzi/strimzi-kafka-operator#6762] [Q&A] [answered]
Is it somehow possible to use secrets in the KafkaConnector config?
I would like to configure a JDBC source connector and need to provide database credentials in `.spec.config.connection.password`,  however this resource is checked-in in version control and I don't want the credentials in plaintext there.

**Comment by scholzj** [comment, marked as answer]
Please check the docs. There is section about using secrets in Kafka Connect in the docs. You can map the secrets there and use the File / Directory configuration providers or you can use the Secret provder directly (but in that case you need to do some additional RBAC setup as well).

**Reply by m-kay** [reply]
Sorry I was not able to find it in the docs before but with your hints I fond this: https://strimzi.io/docs/operators/latest/configuring.html#proc-loading-config-from-config-map-str

Thanks

**Comment by kmcrawford** [comment]
If using an secret from an ENV you can follow this:
https://github.com/strimzi/kafka-env-var-config-provider

if using a secret from a file like aws creds you can follow this:
https://developers.redhat.com/blog/2020/02/14/using-secrets-in-apache-kafka-connect-configuration#building_new_container_images_with_the_connector
